core/chains/evm/client/rpc_client_test.go

```diff
@@ -181,41 +160,74 @@ func TestRPCClient_SubscribeToHeads(t *testing.T) {
 		assert.Equal(t, int64(0), latest.FinalizedBlockNumber)
 		assert.Equal(t, big.NewInt(1000), latest.TotalDifficulty)
 
-		assert.Equal(t, int64(256), highestUserObservations.BlockNumber)
+		assert.Equal(t, int64(0), highestUserObservations.BlockNumber)
 		assert.Equal(t, int64(0), highestUserObservations.FinalizedBlockNumber)
-		assert.Equal(t, big.NewInt(1000), highestUserObservations.TotalDifficulty)
+		assert.Nil(t, highestUserObservations.TotalDifficulty)
 	})
 	t.Run("SubscribeToHeads with http polling enabled will update new heads", func(t *testing.T) {
-		server := testutils.NewWSServer(t, chainId, serverCallBack)
-		wsURL := server.WSURL()
+		type rpcServer struct {
+			Head *evmtypes.Head
+			URL  *url.URL
+		}
+		createRPCServer := func() *rpcServer {
+			server := &rpcServer{}
+			server.URL = testutils.NewWSServer(t, chainId, func(method string, params gjson.Result) (resp testutils.JSONRPCResponse) {
+				assert.Equal(t, "eth_getBlockByNumber", method)
+				if assert.True(t, params.IsArray()) && assert.Equal(t, "latest", params.Array()[0].String()) {
+					head := server.Head
+					jsonHead, err := json.Marshal(head)
+					if err != nil {
+						panic(fmt.Errorf("failed to marshal head: %w", err))
+					}
+					resp.Result = string(jsonHead)
```

Comment:
Why panic? How about just failing the test?
```suggestion
					if assert.NoError(t, err, "failed to marshal head") {
						resp.Result = string(jsonHead)
					}
```

Reply:
It looks like this was moved, rather than newly introduced... but I agree, good time to fix it